Helper/Config.php

```diff
@@ -80,6 +91,43 @@ public function getNotificationsIpHmacCheck($storeId = null)
         );
     }
 
+    /**
+     * Retrieve key for notifications_hmac_key
+     *
+     * @param int $storeId
+     * @return string
+     */
+    public function getNotificationsHmacKey($storeId = null)
+    {
+        $key = "";
+        if ($this->isDemoMode($storeId)) {
+            $key = $this->getConfigData(
+                self::XML_NOTIFICATIONS_HMAC_KEY_TEST,
+                self::XML_ADYEN_ABSTRACT_PREFIX,
+                $storeId,
+                false
+            );
+        } else {
+            $key = $this->getConfigData(
+                self::XML_NOTIFICATIONS_HMAC_KEY_LIVE,
+                self::XML_ADYEN_ABSTRACT_PREFIX,
+                $storeId,
+                false
+            );
+        }
+        return $this->encryptor->decrypt(trim($key));
+    }
+    /**
+     * Returns true if the enviroment is set to test, false for live
+     *
+     * @param int $storeId
+     * @return bool
+     */
+    public function isDemoMode($storeId = null)
+    {
+        return $this->getConfigData('demo_mode', 'adyen_abstract', $storeId, true);
```

Comment:
'adyen_abstract' is already in a constant in this class, and 'demo_mode' can be a new constant to follow the class and method syntax. You can copy getNotificationsIpHmacCheck()

Reply:
Ok I replaced it with the constant